### PR TITLE
Inomurko/231 boba deployer fix

### PR DESCRIPTION
**Pull request by InoMurko**
Closes (https://github.com/bobanetwork/boba/issues/231)

## Overview

Occasionally the boba deployer would die because of hiccups in DAO deployment. 

## Changes

- Grep docker compose logs and detect DAO exit and restart the boba deploy container.

## Testing

/ Should reduce occasional integration test failure.

### CODE REVIEW

**Review by boyuan-chen — APPROVED** [review]

**Comment by boyuan-chen** [comment]
There is one problem when we redeploy the`boba-deployer`. The `CDM Fast` address is changed and it's different from what we have in SDK. The message relayer fast won't work.

**Comment by InoMurko** [comment]
> There is one problem when we redeploy the`boba-deployer`. The `CDM Fast` address is changed and it's different from what we have in SDK. The message relayer fast won't work.

Can you tell me more? How is it that the CDM Fast is registering a stale address?

**Comment by boyuan-chen** [comment]
> > There is one problem when we redeploy the`boba-deployer`. The `CDM Fast` address is changed and it's different from what we have in SDK. The message relayer fast won't work.
> 
> Can you tell me more? How is it that the CDM Fast is registering a stale address?

The CDM Fast address is hardcoded in https://github.com/bobanetwork/boba/blob/6e034ed2817105846a19aac3f50241fade1e77f1/packages/sdk/src/utils/contracts.ts#L127. Redeploying the CDM Fast contract will generate another contract address. The message relayer fast still uses the contract address in SDK, because it uses SDK to query and analyze events.

**Comment by InoMurko** [comment]
> > > There is one problem when we redeploy the`boba-deployer`. The `CDM Fast` address is changed and it's different from what we have in SDK. The message relayer fast won't work.
> 
> > 
> 
> > Can you tell me more? How is it that the CDM Fast is registering a stale address?
> 
> 
> 
> The CDM Fast address is hardcoded in https://github.com/bobanetwork/boba/blob/6e034ed2817105846a19aac3f50241fade1e77f1/packages/sdk/src/utils/contracts.ts#L127. Redeploying the CDM Fast contract will generate another contract address. The message relayer fast still uses the contract address in SDK, because it uses SDK to query and analyze events.

Aha. Yes, makes sense!

File: packages/boba/contracts/deploy/013-BobaDao.deploy.ts [hunk 1]
```diff
@@ -175,24 +172,14 @@ const deployFn: DeployFunction = async (hre) => {
...
-
-  await setPendingAdminTx.wait()
+  let eta1
+  let setPendingAdminData
+  try {
+    ;[eta1, setPendingAdminData] = await setPendingAdmin(hre, eta_delay_s)
+  } catch (error) {
+    console.log(`setPendingAdmin failed because of : ${error}`)
+    ;[eta1, setPendingAdminData] = await setPendingAdmin(hre, eta_delay_s)
+  }
```
Comment (InoMurko) [inline packages/boba/contracts/deploy/013-BobaDao.deploy.ts:182]: @souradeep-das from your experience, does a try-catch catch a failed deployment transaction?
I've had 10 successful runs in cirleci integration tests.it seems so?

Comment (souradeep-das) [reply packages/boba/contracts/deploy/013-BobaDao.deploy.ts:182]: oh yes, a failed tx should go right into the catch block
great, this looks like a smart way to do this for the time being!! 😎

**Review by souradeep-das — APPROVED** [review]

**Review by wsdt — APPROVED** [review]
looks great!

**Pull request merged**